ui/src/components/RenameFlow.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  FormControl,
+  FormLabel,
+  FormErrorMessage,
+  Input,
+  HStack,
+} from "@chakra-ui/react";
+import { ChangeEvent, useContext, useEffect, useState } from "react";
+import { listWorkflows, updateFlow } from "../WorkspaceDB";
+import { RecentFilesContext } from "../WorkspaceContext";
+
+type Props = {
+  handleClose: () => void;
+};
+
+export default function RenameFlow({
+  handleClose,
+}: Props) {
+  const { renameFlowId, onRefreshFilesList } = useContext(RecentFilesContext);
+
+  const [submitError, setSubmitError] = useState("");
+  const [editName, setEditName] = useState("");
+  const previousName = listWorkflows().find(workflow => workflow.id === renameFlowId)?.name;
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setEditName(event.target.value);
+    submitError && setSubmitError("");
+  };
+
+  useEffect(() => {
+    const flow = listWorkflows().find(workflow => workflow.id === renameFlowId);
+
+    setEditName(flow?.name || "")
+  }, [renameFlowId])
+
+  const onSubmit = () => {
+    if (renameFlowId) {
+      const trimEditName = editName.trim();
+      setEditName(trimEditName);
+      if (previousName === trimEditName) return onCloseModal();
+      const flowNameList = listWorkflows()?.map((flow) => flow.name);
+      if (flowNameList.includes(trimEditName)) {
+        setSubmitError(
+          "The name is duplicated, please modify it and submit again."
+        );
+      } else {
+        updateFlow(renameFlowId, {
+          name: trimEditName,
+        });
+        onRefreshFilesList?.();
+        onCloseModal();
+      }
+    }
+  };
+
+  const onCloseModal = () => {
+    setSubmitError("");
+    handleClose();
+  };
+
+  if (renameFlowId === null) return null;
+
+  return (
+    <HStack>
+      <Modal isOpen={renameFlowId !== null} onClose={onCloseModal}>

```

Comment:
make this lazy load by setting this to true and 
{renameFlowId !== null && <RenameFlow />} in parent